starknet_py/tests/e2e/account/outside_execution_test.py

```diff
@@ -124,9 +159,49 @@ async def test_account_outside_execution_for_impossible_time_bounds(
         caller=ANY_CALLER,
     )
 
-    tx = await argent_account.execute_v1(calls=[call], max_fee=MAX_FEE)
+    tx = await account.execute_v1(calls=[call], max_fee=MAX_FEE)
 
     with pytest.raises(TransactionRevertedError) as err:
-        await argent_account.client.wait_for_tx(tx.transaction_hash)
+        await argent_account_v040.client.wait_for_tx(tx.transaction_hash)
 
     assert "argent/invalid-timestamp" in err.value.message
+
+
+@pytest.mark.asyncio
+async def test_account_outside_execution_by_itself_is_impossible(
+    argent_account_v040: BaseAccount,
+    map_contract,
+):
+
+    assert any(
+        [
+            await argent_account_v040.supports_interface(
+                OutsideExecutionInterfaceID.V1
+            ),
+            await argent_account_v040.supports_interface(
+                OutsideExecutionInterfaceID.V2
+            ),
+        ]
+    )
+
+    put_call = Call(
+        to_addr=map_contract.address,
+        selector=get_selector_from_name("put"),
+        calldata=[20, 20],
+    )
+
+    call = await argent_account_v040.sign_outside_execution_call(
+        calls=[put_call],
+        execution_time_bounds=OutsideExecutionTimeBounds(
+            execute_after=datetime.datetime.now() - datetime.timedelta(days=10),
+            execute_before=datetime.datetime.now() - datetime.timedelta(days=9),
+        ),
+        caller=ANY_CALLER,
+    )
+
+    tx = await argent_account_v040.execute_v1(calls=[call], max_fee=MAX_FEE)
```

Comment:
Please use `execute_v3`.

Reply:
Huh! How did those sneak in?! Thank you for catching.